src/number_theory/number_field/embeddings.lean

```diff
@@ -132,39 +116,31 @@ begin
   let C := nat.ceil ((max B 1) ^ (finrank ℚ K) * (finrank ℚ K).choose ((finrank ℚ K) / 2)),
   have := bUnion_roots_finite (algebra_map ℤ K) (finrank ℚ K) (finite_Icc (-C : ℤ) C),
   refine this.subset (λ x hx, _),
-  have h_map_rat_minpoly := minpoly.gcd_domain_eq_field_fractions' ℚ hx.1,
-  rw mem_Union,
-  use minpoly ℤ x,
-  rw [mem_Union, exists_prop],
-  refine ⟨⟨_, λ i, _⟩, _⟩,
-  { rw [← nat_degree_map_eq_of_injective (algebra_map ℤ ℚ).injective_int _, ← h_map_rat_minpoly],
+  have h_map_ℚ_minpoly := minpoly.gcd_domain_eq_field_fractions' ℚ hx.1,
+  simp_rw mem_Union,
+  refine ⟨_, ⟨_, λ i, _⟩, (mem_root_set_iff (minpoly.ne_zero hx.1) x).2 (minpoly.aeval ℤ x)⟩,
+  { rw [← (minpoly.monic hx.1).nat_degree_map (_ : ℤ →+* ℚ), ← h_map_ℚ_minpoly],
     exact minpoly.nat_degree_le (is_integral_of_is_scalar_tower _ hx.1), },
-  { rw [mem_Icc, ← abs_le, ← @int.cast_le ℝ],
-    apply le_trans _ (nat.le_ceil _),
-    convert coeff_bdd_of_norm_le hx.2 i,
-    simp only [h_map_rat_minpoly, coeff_map, eq_int_cast, int.norm_cast_rat, int.norm_eq_abs,
-      int.cast_abs], },
-  { rw [finset.mem_coe, multiset.mem_to_finset, mem_roots, is_root.def, ← eval₂_eq_eval_map,
-      ← aeval_def],
-    { exact minpoly.aeval ℤ x, },
-    { exact monic.ne_zero (monic.map (algebra_map ℤ K) (minpoly.monic hx.1)), }},
+  rw [mem_Icc, ← abs_le, ← @int.cast_le ℝ],
+  apply le_trans _ (nat.le_ceil _),
+  convert coeff_bdd_of_norm_le hx.2 i,
+  rw [h_map_ℚ_minpoly, coeff_map, eq_int_cast, int.norm_cast_rat, int.norm_eq_abs, int.cast_abs],
 end
 
 /-- An algebraic integer whose conjugates are all of norm one is a root of unity. -/
 lemma pow_eq_one_of_norm_eq_one {x : K}
-  (hxi : is_integral ℤ x)  (hx : ∀ φ : K →+* A, ∥φ x∥ = 1) :
+  (hxi : is_integral ℤ x) (hx : ∀ φ : K →+* A, ∥φ x∥ = 1) :
```

Comment:
`goals accomplished` is shown but the kernel fails to check the proof. This is probably a `wlog` bug, [reported here](https://github.com/leanprover-community/mathlib/pull/16495#issuecomment-1266450802).

Reply:
I've now replaced the buggy wlog with a trick, since it doesn't seem like the bug will be fixed soon:
```
replace habne := habne.lt_or_lt,
have : _, swap, cases habne, swap,
{ revert a b, exact this },
```
This also means people working on #16495 won't need to [fix this](https://github.com/leanprover-community/mathlib/pull/16495/files#file-src-number_theory-number_field-embeddings-lean-L161#16495).